couchdb/10-docker-default.ini

```
@@ -19,14 +21,12 @@ socket_options = [{sndbuf, 262144}, {nodelay, true}]
 require_valid_user = true
 
 [httpd]
-port = 5986
-bind_address = 0.0.0.0
 secure_rewrites = false
 max_http_request_size = 134217728 ; 128 MiB
 WWW-Authenticate = Basic realm="Medic Mobile Web Services"
 
 [couch_httpd_auth]
-# timeout is set to 1 year in seconds, 31536000
+; timeout is set to 1 year in seconds, 31536000
 timeout = 31536000
```

would be great to expose prometheus monitoring via an unauthenticated endpoint https://github.com/medic/cht-watchdog/issues/76.  happy to do this in a second PR after this merges

---

Can I suggest you develop a user configurable config layer? We're developing against this and having to modify directly means it's difficult to stay conformant with upstream. 

See for example:

```suggestion
echo "
[daemons]
httpsd = {couch_httpd, start_link, [https]}
[admins]
admin = ${COUCHDB_PASSWORD}
[ssl]
port = 6984
enable = true
cert_file = /opt/couchdb/etc/certs/cert.pem
key_file = /opt/couchdb/etc/certs/privkey.pem
cacert_file = /opt/couchdb/etc/certs/chain.pem" |

sudo crudini --merge /opt/couchdb/etc/local.ini
```

---

Can I suggest you develop a user configurable config layer? We're developing against this and having to modify directly means it's difficult to stay conformant with upstream. 

See for example:

```suggestion
echo "
[daemons]
httpsd = {couch_httpd, start_link, [https]}
[admins]
admin = ${COUCHDB_PASSWORD}
[ssl]
port = 6984
enable = true
cert_file = /opt/couchdb/etc/certs/cert.pem
key_file = /opt/couchdb/etc/certs/privkey.pem
cacert_file = /opt/couchdb/etc/certs/chain.pem" |

sudo crudini --merge /opt/couchdb/etc/local.ini
```

---

Hi @fardarter 

Thanks a lot for this great suggestion! 
I think this is out of scope of this PR, and this has been discussed while we originally developed 4.x. We have not reached a conclusion, and we decided this would be a goal for later. 

Would you mind creating an issue that details exactly which configs you need and how you're thinking about persisting them in the container? 
Thanks a lot!

---

Our current plan is to build our own version of the images with patches as needed, so this would be a pre-image build step. Would argue that supporting user patching behaviour would help with keeping people contributing to upstream.

I've set up a system where we're doing our best to only patch what we need. Could be improved; happy to demo it.

I'm actually trying to find your contact details at the moment (which is why I'm on this PR). Don't suppose I could give you mine (and then we could delete the comment)?

My senior dev on this project has fallen ill and he was using this PR as a way to let us use 3.3.2 (we'd like to use the bitnami image on azure, but I can script those differences).

I'm supposed to be doing ops so I'm a little lost as to whether we can even use 3.3.2 at this juncture, or whether it would be foolish. I'd prefer not having to build out a 2.2.1 on a VM where there's a preconfigured, hardened bitnami image. 

I'm also a little lost as to what the right command is to test this locally, but my tests seem to be breaking.

---

Open discussion is welcome and desirable on this PR or on the [CHT Forum](https://forum.communityhealthtoolkit.org/). 
If you have specific questions, or have a technical challenge, the forum would be the best place to express these, so that other community members would benefit. 
I believe there already is a thread about this PR: https://forum.communityhealthtoolkit.org/t/upgrade-of-couchdb-from-version-2-3-1-to-3-3-1/2928 , I'm not sure if this is at all related to your issue. 

Please feel free to reach out on either platform! 
I'm really keen to hear more about your desired outcome. 

Thanks!